app/src/behaviors/behavior_backlight.c

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2020 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#define DT_DRV_COMPAT zmk_behavior_backlight
+
+#include <device.h>
+#include <drivers/behavior.h>
+#include <logging/log.h>
+
+#include <dt-bindings/zmk/bl.h>
+#include <zmk/backlight.h>
+#include <zmk/keymap.h>
+
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+#if DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT)
+
+static int behavior_backlight_init(const struct device *dev) { return 0; }
+
+static int on_keymap_binding_pressed(struct zmk_behavior_binding *binding,
+                                     struct zmk_behavior_binding_event event) {
+    LOG_DBG("Backlight command: %d", binding->param1);
+    switch (binding->param1) {
+    case BL_TOG:
+        return zmk_backlight_toggle();
+    case BL_ON:
+        return zmk_backlight_on();
+    case BL_OFF:
+        return zmk_backlight_off();
+    case BL_INC:
+        return zmk_backlight_inc();
+    case BL_DEC:
+        return zmk_backlight_dec();
+    default:
+        LOG_ERR("Unknown backlight command: %d", binding->param1);
+    }
+
+    return -ENOTSUP;
+}
+
+static int on_keymap_binding_released(struct zmk_behavior_binding *binding,
+                                      struct zmk_behavior_binding_event event) {
+    return ZMK_BEHAVIOR_OPAQUE;
```

Comment:
Not something to change in this commit, but it seems a bit silly that we require a released callback for behaviors that only do something on press.

@petejohanson maybe we could change `z_impl_behavior_keymap_binding_released` so it returns `ZMK_BEHAVIOR_OPAQUE` instead of `-ENOTSUP` if no release callback is set?

Reply:
I think perhaps returning that is still valid, but the keymap code could better handle that errno specifically.